tests/main/hidden-snap-dir/task.yaml

```diff
@@ -184,3 +184,22 @@ execute: |
     check_env "--with-exposed-home" "x3"
     check_dirs "--with-exposed-home" "x3"
     check_data "--with-exposed-home" "x3" "$data"
+
+    echo "Check migration after remove works"
+    # ensure dirs under ~/.snap/data are created
+    "$NAME".cmd 'true'
+
+    echo "Remove snap and check dir was cleaned up"
+    snap remove --purge "$NAME"
+    not test -e "$HOME/.snap/data/$NAME"
+
+    echo "Install snap under ~/snap"
+    snap pack "$TESTSLIB/snaps/$NAME"
+    snap unset system experimental.hidden-snap-folder
+    "$TESTSTOOLS"/snaps-state install-local "$NAME"
+    # create dirs under ~/snap to be migrated
+    "$NAME".cmd 'true'
```

Comment:
Check for `test -e "$HOME/snap/$NAME"` after this line

Reply:
+1